### DISCUSSION TITLE
[BUG]  std::runtime_error isn't passed to Python?

### DISCUSSION THREAD

**Discussion by yurivict** [pybind/pybind11#4289] [General]
When C++ code throws a std::runtime_error exception python catches it as ```RuntimeError``` that prints as  ```Caught an unknown exception!```

The documentation says that a variety of STL exceptions is caught and converted to Python exceptions, but ```std::runtime_error``` isn't listed there.

I have 2 questions:
(1) Why isn't ```std::runtime_error``` listed as a handled exception type?
(2) Why isn't ```std::runtime_error``` processed as an instance of ```std::exception``` properly?

**Comment by Skylion007** [comment]
@yurivict Do you have a minimal example? We can't replicate this and we even have testing to ensure the proper string is passed to Python when it is thrown: https://github.com/pybind/pybind11/blob/fcb5554d9fdd4ad3150bf12cb8be04099cb4976e/tests/test_exceptions.py#L10

**Comment by youben11** [comment]
I'm observing this on MacOS 12 on an M1 chip. Tried with both `std::runtime_error` and `std::invalid_argument`, they both got caught as `RuntimeError: Caught an unknown exception!` on the python side.

**Comment by GenieTim** [comment]
A year ago, I did not have this problem, but I am also observing it now on my Intel-based MacOS system.

**Comment by linusheck** [comment]
I'm also observing this on an M1 chip.
